thinc/backends/ops.py

```diff
@@ -680,6 +685,191 @@ def backprop_relu(
+        if inplace:
+            dY *= dX
+            return dY
+        return dY * dX
+
+    def relu_k(
+        self, X: FloatsType, n: float = 6.0, inplace: bool = False
+    ) -> FloatsType:
+        return self.clipped_linear(X, max_val=n, inplace=inplace)
+
+    def backprop_relu_k(
+        self, dY: FloatsType, X: FloatsType, n: float = 6.0, inplace: bool = False
+    ) -> FloatsType:
+        return self.backprop_clipped_linear(dY, X, max_val=n, inplace=inplace)
+
+    def hard_sigmoid(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        return self.clipped_linear(X, slope=0.2, offset=0.5)
+
+    def backprop_hard_sigmoid(
+        self, dY: FloatsType, X: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        return self.backprop_clipped_linear(dY, X, slope=0.2, offset=0.5)
+
+    def hard_tanh(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        return self.clipped_linear(X, min_val=-1.0, max_val=1.0)
+
+    def backprop_hard_tanh(
+        self, dY: FloatsType, X: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        return self.backprop_clipped_linear(dY, X, min_val=-1.0, max_val=1.0)
+
+    def swish(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        if inplace:
+            X *= self.sigmoid(X)  # type: ignore
+            return cast(FloatsType, X)
+        out = X * self.sigmoid(X)  # type: ignore
+        return cast(FloatsType, out)
+
+    def backprop_swish(
+        self, dY: FloatsType, X: FloatsType, Y: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        Y = Y + self.sigmoid(X) * (1 - Y)  # type: ignore
+        if inplace:
+            dY *= Y  # type: ignore
+            return cast(FloatsType, dY)
+        out = dY * Y  # type: ignore
+        return cast(FloatsType, out)
+
+    # Following https://www.scitepress.org/Papers/2019/74696/74696.pdf
+    def hard_swish(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        if inplace:
+            X *= self.hard_sigmoid(X)  # type: ignore
+            return cast(FloatsType, X)
+        out = X * self.hard_sigmoid(X)  # type: ignore
+        return cast(FloatsType, out)
+
+    def backprop_hard_swish(
+        self, dY: FloatsType, X: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        dX = X * 0.4 + 0.5
+        dX[X > 2.5] = 1.0
+        dX[X < -2.5] = 0
+        if inplace:
+            dY *= dX
+            return dY
+        return dY * dX
+
+    # From https://arxiv.org/pdf/1905.02244v5.pdf
+    def hard_swish_mobilenet(self, X: FloatsType, inplace: bool = False) -> FloatsType:
+        if inplace:
+            X *= self.relu_k(X + 3) / 6
+            return X
+        return X * (self.relu_k(X + 3) / 6)
+
+    def backprop_hard_swish_mobilenet(
+        self, dY: FloatsType, X: FloatsType, inplace: bool = False
+    ) -> FloatsType:
+        dX = (1 / 6) * (X * 2.0 + 3.0)
+        dX[X > 3.0] = 1.0
+        dX[X < -3.0] = 0
```

Comment:
This looks like it'll return `float64` I think

Reply:
Seems like it, but it passes the test for some reason.